File: JAVA/Minimum Absolute Difference In BST.java
class Solution
{
    
    int prev;
    int diff;
    
    int absolute_diff(Node root)
    {
        prev = Integer.MIN_VALUE;
        diff = Integer.MAX_VALUE;
        
        findMinDiff(root);
        return diff;
    }
    
    void findMinDiff(Node root){
        if(root == null)
            return;
            
        findMinDiff(root.left);
        if(prev!= Integer.MIN_VALUE && Math.abs(prev-root.data) < diff){
            diff = Math.abs(prev-root.data);
        }
        prev = root.data;
        
        findMinDiff(root.right);
    }
}

File: C++/Minimum Absolute Difference In BST.cpp
class Solution
{
    public:
    vector<int>res;
    void Inorder(Node *root)
    {
        if(root == NULL)
            return ;
            Inorder(root->left);
            res.push_back(root->data);
            Inorder(root->right);
    }
    int absolute_diff(Node *root)
    {
        Inorder(root);
        int min = INT_MAX;
        for(int i=res.size()-1;i>0;i--)
        {
            if(res[i]-res[i-1] < min)
            min = res[i]-res[i-1];
        }
        return min;
    }
};

